Implement scrollbar to list

JustinBeBoy opened the pull request:
Resolve #609
- Upgrade gioui new version
- Implement scrollbar for list
- Update current component for gioui new version
- Custom Icon

![scroll1](https://user-images.githubusercontent.com/19331824/136903488-24bee0a3-8e07-498d-878e-e75f1edb59fc.PNG)

![scroll2](https://user-images.githubusercontent.com/19331824/136903410-7b216665-6d38-4a10-b358-5005e8c9c231.PNG)

![scroll3](https://user-images.githubusercontent.com/19331824/136903441-7af2ce10-5cf5-4790-94f0-22ee70df5821.PNG)

JustinBeBoy committed:
add custom list with scroll view

Sirmorrison commented:
I dont think you will need those custom widgets you are adding since the recent updates to the gioui package has a scrollbar widget attached.
https://github.com/gioui/gio/blob/main/widget/list.go
Correct me if i am wrong but i believe this is what we want.

JustinBeBoy committed:
update gio version and refactor code with new gioui version

Sirmorrison commented:
Close #665

Sirmorrison commented:
Kindly rebase this Pr as the gioui library has been updated to a more recent version.

also, from the first screenshot your dropped, the scrollbar widget does not align with the page content area.. that the the area that is to scroll.
<img width="280" alt="image" src="https://user-images.githubusercontent.com/27733432/137122990-60118b65-03a2-4192-bcef-8c3395429e1b.png">

The second screenshot aligns well to the scroll area, however, it is too close to the content card.. there should be some padding in-between.
<img width="93" alt="image" src="https://user-images.githubusercontent.com/27733432/137123358-bd73a2b7-022e-42e7-93e8-63fe629e71c2.png">

The 3rd screenshot has it scroll bar inside the content area. It should be outside like the others.
<img width="228" alt="image" src="https://user-images.githubusercontent.com/27733432/137123561-e2cbe065-df1b-4e59-9fb9-b72c6cb48dd1.png">

JustinBeBoy committed:
rebase and fix scroll bar in page used

Sirmorrison commented on ui/decredmaterial/icon.go:
```diff
@@ -3,20 +3,32 @@
 type Icon struct {
 	*widget.Icon
+	Color color.NRGBA
+	Size  int
```
This is redundant. Kindly remove.

Sirmorrison commented on ui/decredmaterial/icon.go:
```diff
@@ -3,20 +3,32 @@
-	return &Icon{icon}, nil
+}
+
+func (icon *Icon) Layout(gtx C) D {
```
```suggestion
func (icon *Icon) Layout(gtx C, iconSize unit.Value) D {
```

Sirmorrison commented on ui/decredmaterial/icon.go:
```diff
@@ -3,20 +3,32 @@
+	if icon.Color != (color.NRGBA{}) {
+		cl = icon.Color
+	}
+	if icon.Size != 0 {
+		gtx.Constraints.Min.X = icon.Size
+	}
```
```suggestion
	gtx.Constraints.Min.X = gtx.Px(iconSize)
```

Sirmorrison commented on ui/modal/info_modal.go:
```diff
@@ -189,8 +187,9 @@ func (in *InfoModal) Layout(gtx layout.Context) D {
-				gtx.Constraints.Min.X = gtx.Px(values.MarginPadding50)
-				return in.dialogIcon.Layout(gtx, in.Theme.Color.DeepBlue)
+				in.dialogIcon.Color = in.Theme.Color.DeepBlue
+				in.dialogIcon.Size = 50
+				return in.dialogIcon.Layout(gtx)
```
```suggestion
				return in.dialogIcon.Layout(gtx, values.MarginPadding50)
```

Sirmorrison commented on ui/modal/info_modal_layouts.go:
```diff
@@ -51,9 +51,10 @@ func privacyInfo(th *decredmaterial.Theme) []layout.Widget {
-					return ic.Layout(gtx, th.Color.Gray)
+					ic := decredmaterial.NewIcon(decredmaterial.MustIcon(widget.NewIcon(icons.ImageLens)))
+					ic.Color = th.Color.Gray
+					ic.Size = 8
+					return ic.Layout(gtx)
```
```suggestion
					return ic.Layout(gtx, values.MarginPadding8)
```

Sirmorrison commented on ui/modal/info_modal_layouts.go:
```diff
@@ -70,9 +71,10 @@ func privacyInfo(th *decredmaterial.Theme) []layout.Widget {
-					return ic.Layout(gtx, th.Color.Gray)
+					ic := decredmaterial.NewIcon(decredmaterial.MustIcon(widget.NewIcon(icons.ImageLens)))
+					ic.Color = th.Color.Gray
+					ic.Size = 8
+					return ic.Layout(gtx)
```
```suggestion
					return ic.Layout(gtx, values.MarginPadding8)
```

Sirmorrison commented on ui/modal/text_input_modal.go:
```diff
@@ -139,11 +139,12 @@ func (tm *TextInputModal) Layout(gtx layout.Context) D {
 					return inset.Layout(gtx, func(gtx C) D {
-						gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-						return img.Layout(gtx, tm.Theme.Color.Gray3)
+						img.Size = 20
+						return img.Layout(gtx)
```
```suggestion
						return img.Layout(gtx, values.MarginPadding20)
```

Sirmorrison commented on ui/page/about_page.go:
```diff
@@ -121,8 +119,8 @@ func (pg *AboutPage) layoutRows(gtx layout.Context) layout.Dimensions {
 							return in.Layout(gtx, func(gtx C) D {
-								gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-								return pg.chevronRightIcon.Layout(gtx, pg.Theme.Color.Gray)
+								pg.chevronRightIcon.Size = 20
+								return pg.chevronRightIcon.Layout(gtx)
```
```suggestion
								return pg.chevronRightIcon.Layout(gtx, values.MarginPadding20)
```

Sirmorrison commented on ui/page/components/account_selector.go:
```diff
@@ -167,8 +167,9 @@ func (as *AccountSelector) Layout(gtx layout.Context) layout.Dimensions {
-							gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-							return as.Icons.DropDownIcon.Layout(gtx, as.Theme.Color.Gray3)
+							ic := decredmaterial.NewIcon(as.Icons.DropDownIcon)
+							ic.Size = 20
+							return ic.Layout(gtx)
```
```suggestion
							return ic.Layout(gtx, values.MarginPadding20)
```

Sirmorrison commented on ui/page/components/account_selector.go:
```diff
@@ -434,8 +435,9 @@ func (asm *AccountSelectorModal) walletAccountLayout(gtx layout.Context, account
-						gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-						return asm.Icons.NavigationCheck.Layout(gtx, asm.Theme.Color.Gray3)
+						ic := decredmaterial.NewIcon(asm.Icons.NavigationCheck)
+						ic.Size = 20
+						return ic.Layout(gtx)
```
```suggestion
						return ic.Layout(gtx,values.MarginPadding20)
```

Sirmorrison commented on ui/page/debug_page.go:
```diff
@@ -79,8 +78,9 @@ func (pg *DebugPage) debugItem(gtx C, i int) D {
-					gtx.Constraints.Min.X = gtx.Px(values.MarginPadding22)
-					return pg.Icons.ChevronRight.Layout(gtx, pg.Theme.Color.Gray)
+					ic := decredmaterial.NewIcon(pg.Icons.ChevronRight)
+					ic.Size = 22
+					return ic.Layout(gtx)
```
```suggestion
					return ic.Layout(gtx, values.MarginPadding22)
```

Sirmorrison commented on ui/page/overview_page.go:
```diff
@@ -384,9 +387,10 @@ func (pg *OverviewPage) blockInfoRow(gtx layout.Context) layout.Dimensions {
 		}),
 		layout.Rigid(func(gtx C) D {
+			pg.walletStatusIcon.Color = pg.Theme.Color.Gray
+			pg.walletStatusIcon.Size = 5
```
```suggestion
```

Sirmorrison commented on ui/page/overview_page.go:
```diff
@@ -384,9 +387,10 @@ func (pg *OverviewPage) blockInfoRow(gtx layout.Context) layout.Dimensions {
 			return layout.Inset{Right: values.MarginPadding5, Top: values.MarginPadding8}.Layout(gtx, func(gtx C) D {
-				gtx.Constraints.Min.X = gtx.Px(values.MarginPadding5)
-				return pg.walletStatusIcon.Layout(gtx, pg.Theme.Color.Gray)
+				return pg.walletStatusIcon.Layout(gtx)
```
```suggestion
				return pg.walletStatusIcon.Layout(gtx, values.MarginPadding5)
```

Sirmorrison commented on ui/page/overview_page.go:
```diff
@@ -416,18 +420,19 @@ func (pg *OverviewPage) syncBoxTitleRow(gtx layout.Context) layout.Dimensions {
 	if pg.isConnnected {
 		statusLabel.Text = values.String(values.StrOnline)
-		clr = pg.Theme.Color.Success
+		//clr = pg.Theme.Color.Success
```
Remove if no longer needed.

Sirmorrison commented on ui/page/overview_page.go:
```diff
@@ -416,18 +420,19 @@ func (pg *OverviewPage) syncBoxTitleRow(gtx layout.Context) layout.Dimensions {
 	statusLabel := pg.Theme.Body1(values.String(values.StrOffline))
-	clr := pg.Theme.Color.Danger
+	pg.walletStatusIcon.Color = pg.Theme.Color.Danger
+	pg.walletStatusIcon.Size = 14
```
```suggestion
```

Sirmorrison commented on ui/page/overview_page.go:
```diff
@@ -416,18 +420,19 @@ func (pg *OverviewPage) syncBoxTitleRow(gtx layout.Context) layout.Dimensions {
 				return layout.Inset{Top: values.MarginPadding4, Right: values.MarginPadding4}.Layout(gtx, func(gtx C) D {
-					gtx.Constraints.Min.X = gtx.Px(values.MarginPadding14)
-					return pg.walletStatusIcon.Layout(gtx, clr)
+					return pg.walletStatusIcon.Layout(gtx)
```
```suggestion
					return pg.walletStatusIcon.Layout(gtx, values.MarginPadding14)
```

Sirmorrison commented on ui/page/overview_page.go:
```diff
@@ -468,8 +473,9 @@ func (pg *OverviewPage) syncStatusTextRow(gtx layout.Context, inset layout.Inset
-							gtx.Constraints.Min.X = gtx.Px(values.MarginPadding16)
-							return pg.cachedIcon.Layout(gtx, pg.Theme.Color.Gray)
+							pg.cachedIcon.Color = pg.Theme.Color.Gray
+							pg.cachedIcon.Size = 16
+							return pg.cachedIcon.Layout(gtx)
```
```suggestion
							return pg.cachedIcon.Layout(gtx, values.MarginPadding16)
```

Sirmorrison commented on ui/page/overview_page.go:
```diff
@@ -491,18 +497,18 @@ func (pg *OverviewPage) syncStatusTextRow(gtx layout.Context, inset layout.Inset
 	return i.Layout(gtx, func(gtx C) D {
-		gtx.Constraints.Min.X = gtx.Px(values.MarginPadding24)
-		return syncStatusIcon.Layout(gtx, col)
+		syncStatusIcon.Size = 24
+		return syncStatusIcon.Layout(gtx)
```
```suggestion
		return syncStatusIcon.Layout(gtx, values.MarginPadding24)
```

Sirmorrison commented on ui/page/proposal/proposal_details_page.go:
```diff
@@ -59,12 +58,13 @@ func newProposalDetailsPage(l *load.Load, proposal *dcrlibwallet.Proposal) *prop
-		timerIcon:          l.Icons.TimerIcon,
-		viewInPoliteiaBtn:  l.Theme.NewClickable(true),
-		voteBar:            NewVoteBar(l),
+		//voteBar:            l.Theme.VoteBar(decredmaterial.NewIcon(l.Icons.ActionInfo), decredmaterial.NewIcon(l.Icons.ImageBrightness1)),
```
Remove if no longer needed

Sirmorrison commented on ui/page/proposal/proposal_details_page.go:
```diff
@@ -190,14 +190,15 @@ func (pg *proposalDetails) layoutInDiscussionState(gtx C) D {
 				}.Layout(gtx, func(gtx C) D {
-					gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-					return icon.Layout(gtx, pg.Theme.Color.Primary)
+					icon.Size = 24
+					return icon.Layout(gtx)
```
Same as above

Sirmorrison commented on ui/page/proposal/proposal_details_page.go:
```diff
@@ -267,8 +266,8 @@ func (pg *proposalDetails) layoutNormalTitle(gtx C) D {
 					}
-					gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-					return icon.Layout(gtx, iconColor)
+					icon.Size = 20
+					return icon.Layout(gtx)
```
Same as above

Sirmorrison commented on ui/page/proposal/proposals_page.go:
```diff
@@ -493,8 +501,8 @@ func (pg *ProposalsPage) layoutAuthorAndDate(gtx C, item proposalItem) D {
 									pg.layoutInfoTooltip(gtx, rect, item)
-									gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-									return pg.infoIcon.Layout(gtx, pg.Theme.Color.Gray)
+									pg.infoIcon.Size = 20
+									return pg.infoIcon.Layout(gtx)
```
Same as above

Sirmorrison commented on ui/page/proposal/proposals_page.go:
```diff
@@ -551,8 +559,8 @@ func (pg *ProposalsPage) layoutProposalVoteBar(gtx C, item proposalItem) D {
 		layout.Rigid(func(gtx C) D {
-			gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-			return pg.updatedIcon.Layout(gtx, pg.Theme.Color.Success)
+			pg.updatedIcon.Size = 20
+			return pg.updatedIcon.Layout(gtx)
```
Same as above

Sirmorrison commented on ui/page/proposal/votebar_widget.go:
```diff
@@ -232,8 +231,8 @@ func (v *VoteBar) layoutIconAndText(gtx C, lbl decredmaterial.Label, count float
 				return layout.Inset{Right: values.MarginPadding5, Top: values.MarginPadding5}.Layout(gtx, func(gtx C) D {
-					gtx.Constraints.Min.X = gtx.Px(values.MarginPadding10)
-					return v.legendIcon.Layout(gtx, clr)
+					v.legendIcon.Size = 10
+					return v.legendIcon.Layout(gtx)
```
Same as above

Sirmorrison commented on ui/page/proposal/votebar_widget.go:
```diff
@@ -271,8 +270,8 @@ func (v *VoteBar) layoutInfo(gtx C) D {
 			return layout.Inset{Left: values.MarginPadding5}.Layout(gtx, func(gtx C) D {
-				gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-				return v.infoIcon.Layout(gtx, v.Theme.Color.Gray)
+				v.infoIcon.Size = 20
+				return v.infoIcon.Layout(gtx)
```
Same as above

Sirmorrison commented on ui/page/proposal/wallet_selector.go:
```diff
@@ -142,8 +142,9 @@ func (as *WalletSelector) Layout(gtx layout.Context) layout.Dimensions {
-										gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-										return as.Icons.DropDownIcon.Layout(gtx, as.Theme.Color.Gray3)
+										ic := decredmaterial.NewIcon(as.Icons.DropDownIcon)
+										ic.Size = 20
+										return ic.Layout(gtx)
```
Same as above

Sirmorrison commented on ui/page/proposal/wallet_selector.go:
```diff
@@ -319,8 +320,9 @@ func (asm *WalletSelectorModal) walletAccountLayout(gtx layout.Context, wallet *
-									gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-									return asm.Icons.NavigationCheck.Layout(gtx, asm.Theme.Color.Gray3)
+									ic := decredmaterial.NewIcon(asm.Icons.NavigationCheck)
+									ic.Size = 20
+									return ic.Layout(gtx)
```
Same as above

Sirmorrison commented on ui/page/seedbackup/backup_success.go:
```diff
@@ -60,9 +60,9 @@ func (pg *BackupSuccessPage) Layout(gtx C) D {
-						gtx.Constraints.Min.X = gtx.Px(values.MarginPadding24)
-						return successIcon.Layout(gtx, pg.Theme.Color.Success)
+						successIcon := decredmaterial.NewIcon(pg.Icons.ActionCheckCircle)
+						successIcon.Size = 64
+						return successIcon.Layout(gtx)
```
Same as above

Sirmorrison commented on ui/page/send/send_confirm_modal.go:
```diff
@@ -155,10 +155,11 @@ func (scm *sendConfirmModal) Layout(gtx layout.Context) D {
 							return layout.Inset{Right: values.MarginPadding8}.Layout(gtx, func(gtx C) D {
-								gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-								return icon.Layout(gtx, scm.Theme.Color.Gray3)
+								icon.Size = 15
+								return icon.Layout(gtx)
```
Same as above

Sirmorrison commented on ui/page/send/send_confirm_modal.go:
```diff
@@ -230,10 +231,11 @@ func (scm *sendConfirmModal) Layout(gtx layout.Context) D {
 					return layout.Inset{Right: values.MarginPadding10}.Layout(gtx, func(gtx C) D {
-						gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-						return icon.Layout(gtx, scm.Theme.Color.Gray)
+						icon.Size = 20
+						return icon.Layout(gtx)
```
Same as above

Sirmorrison commented on ui/page/settings_page.go:
```diff
@@ -347,8 +346,8 @@ func (pg *SettingsPage) clickableRow(gtx layout.Context, row row) layout.Dimensi
 					layout.Rigid(func(gtx C) D {
-						gtx.Constraints.Min.X = gtx.Px(values.MarginPadding22)
-						return row.icon.Layout(gtx, pg.Theme.Color.LightGray)
+						row.icon.Size = 22
+						return row.icon.Layout(gtx)
```
Same as above

Sirmorrison commented on ui/page/tickets/utils.go:
```diff
@@ -470,8 +470,9 @@ func ticketCard(gtx layout.Context, l *load.Load, tx *transactionItem, showWalle
-										gtx.Constraints.Min.X = gtx.Px(values.MarginPadding5)
-										return l.Icons.ImageBrightness1.Layout(gtx, l.Theme.Color.Gray2)
+										ic := decredmaterial.NewIcon(l.Icons.ImageBrightness1)
+										ic.Size = 5
+										return ic.Layout(gtx)
```
Same as above

Sirmorrison commented on ui/page/tickets/vsp_selector.go:
```diff
@@ -107,8 +107,9 @@ func (v *vspSelector) Layout(gtx layout.Context) layout.Dimensions {
-										gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-										return v.Icons.DropDownIcon.Layout(gtx, v.Theme.Color.Gray3)
+										ic := decredmaterial.NewIcon(v.Icons.DropDownIcon)
+										ic.Size = 20
+										return ic.Layout(gtx)
```
Same as above

JustinBeBoy committed:
refactor Size in Icon

Sirmorrison commented on ui/page/wallet_page.go:
```diff
@@ -357,11 +357,17 @@ func (pg *WalletPage) Layout(gtx layout.Context) layout.Dimensions {
+			return pg.layoutAddWalletSection(gtx)
+		}),
+	)
+	//pg.layoutAddWalletSection(gtx)
```
If not needed, kindly remove.

Sirmorrison commented on ui/page/wallet_page.go:
```diff
@@ -357,11 +357,17 @@ func (pg *WalletPage) Layout(gtx layout.Context) layout.Dimensions {
-			layout.Stacked(func(gtx C) D {
-				return pg.layoutAddWalletSection(gtx)
-			}),
+			//layout.Stacked(func(gtx C) D {
+			//	return pg.layoutAddWalletSection(gtx)
+			//}),
```
If not needed, kindly remove.

Sirmorrison commented on ui/decredmaterial/clickable_list.go:
```diff
@@ -53,31 +66,41 @@ func (cl *ClickableList) handleClickables(count int) {
 		})
+	}
+	gtx.Constraints.Min.X = gtx.Constraints.Max.X
+	return material.List(cl.theme.Base, cl.list).Layout(gtx, count, func(gtx C, i int) D {
```
Why use `gio material` directly? Kindly add this as a file on the `decredmaterial` package. `list.go`.
See here for examples.
https://github.com/planetdecred/godcr/blob/master/ui/decredmaterial/checkbox.go.
https://github.com/planetdecred/godcr/blob/master/ui/decredmaterial/label.go.

Sirmorrison commented on ui/page/seedbackup/verify_seed.go:
```diff
@@ -198,26 +207,23 @@ func (pg *VerifySeedPage) Layout(gtx C) D {
-						return pg.seedList.Layout(gtx, len(pg.multiSeedList), func(gtx C, index int) D {
-							return pg.seedListRow(gtx, index, pg.multiSeedList[index])
+						gtx.Constraints.Min.X = gtx.Constraints.Max.X
+						return material.List(pg.Theme.Base, pg.list).Layout(gtx, len(pg.multiSeedList), func(gtx C, i int) D {
```
Why are we using `material.List`? Kindly add this as a file on the decredmaterial package. list.go.
See here for examples.
https://github.com/planetdecred/godcr/blob/master/ui/decredmaterial/checkbox.go.
https://github.com/planetdecred/godcr/blob/master/ui/decredmaterial/label.go.

Sirmorrison commented on ui/page/tickets/overview.go:
```diff
@@ -132,20 +140,28 @@ func (pg *Page) loadPageData() {
+	}
+	gtx.Constraints.Min.X = gtx.Constraints.Max.X
+	return layout.Inset{Top: values.MarginPadding24}.Layout(gtx, func(gtx layout.Context) layout.Dimensions {
+		return material.List(pg.Theme.Base, pg.list).Layout(gtx, len(widgets), func(gtx C, i int) D {
```
Why use gio material directly? Kindly add this as a file on the decredmaterial package. list.go.
See here for examples.
https://github.com/planetdecred/godcr/blob/master/ui/decredmaterial/checkbox.go.
https://github.com/planetdecred/godcr/blob/master/ui/decredmaterial/label.go.

Sirmorrison reviewed (changes requested):
Scrollbar not visible on the Overview page.
![image](https://user-images.githubusercontent.com/27733432/138266872-715d5975-13b5-4076-b71d-0d52e94fb4da.png)

No scroll bar on tx details page.
![image](https://user-images.githubusercontent.com/27733432/138267104-5c336267-38be-4092-bd14-06f004948378.png)

No scroll bar on wallet page when all the wallets are expanded.
![image](https://user-images.githubusercontent.com/27733432/138267282-603a2df7-bb79-414e-9b68-7e2c5c64a07f.png)

Not so sure why this icon is on the left of your page instead of the right.
![image](https://user-images.githubusercontent.com/27733432/138267523-65e46b2a-892b-4d92-bb0d-5decb4ed473c.png)

The scroll padding on the proposals page is not uniform as the against tx and ticket pages.
![image](https://user-images.githubusercontent.com/27733432/138267863-d6df7e23-69f4-4c6d-aad0-9ba1d94d4f25.png)

Proposal details content area does not have a scroll bar.
![image](https://user-images.githubusercontent.com/27733432/138268088-1e47978a-b833-4985-9d07-d0c3137d88a9.png)

Setting page does not have a scroll bar.
![image](https://user-images.githubusercontent.com/27733432/138268171-063a1e1c-3362-4a52-91f3-0536992a6000.png)

* Wallet Log Page and statistics page do not also have scroll bars.

The idea of this PR is to add scroll bars to all pages that scrolls. Meaning any page that has the `list` layout.

dreacot commented on ui/decredmaterial/clickable_list.go:
```diff
@@ -15,15 +18,25 @@ type ClickableList struct {
 		List:           layout.List{Axis: axis},
 		ClickHighlight: t.Color.SurfaceHighlight,
 		selectedItem:   -1,
+		HideScroll:     false,
```
`isScrollBarHidden` seems like a better variable name

Sirmorrison committed:
fix add wallet menu btn displacement
- add list_scroll widget to decred material
- rename HideScroll to IsHiddenScrollBer

Sirmorrison committed:
Add scroll bar to overview page, wallet page,
- wallet seed backup page and fix seed view page scroll bug
- add scrollbar to tx details and account details pages
- add scroll to ticket page, remove ticket gridlayout

Sirmorrison committed:
Fix see all icon size bug
- add scrollbar to proposals details page,
- license page, log page, statistics page

Sirmorrison committed:
Remove scrollbar from clickable list widget.
- clean up tx page alignment
- ensure uniform scrollbar padding
- rebase PR and clean up code

Sirmorrison committed:
Add scrollbar to send and receive page
- Add scrollbar to modal widget
- fix wallet and ticket page scrollbar bug
